src/Controller/JobsController.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use DateTime;
+use Exception;
+use App\Entity\Jobs;
+use App\Repository\JobsRepository;
+use App\Repository\CompanyRepository;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+/**
+ * JobsController
+ */
+class JobsController extends AbstractController
+{
+    private JobsRepository $jobsRepository;
+    private CompanyRepository $companyRepository;
+
+    /**
+     * __construct
+     *
+     * @param  mixed $jobsRepository
+     * @param  mixed $companyRepository
+     * @return void
+     */
+    public function __construct(JobsRepository $jobsRepository, CompanyRepository $companyRepository)
+    {
+        $this->jobsRepository = $jobsRepository;
+        $this->companyRepository = $companyRepository;
+    }
+
+    /**
+     * addjobs
+     *
+     * @param  mixed $request
+     * @return Response
+     */
+    public function addJob(Request $request): Response
+    {
+        try {
+            $name = $request->get("name");
+            $active = (int)$request->get('active');
+            $description = $request->get("description");
+            $priority = (int)$request->get('priority');
+            $companyId = (int)$request->get('company_id');
+
+            $company =  $this->companyRepository->find($companyId);
+
+            if (is_null($company)) {
+                throw new \InvalidArgumentException('Could not find company with id: ' . $companyId);
+            }
+
+            //create new Jobs objec
+            $job = new Jobs();
+            $job->setName($name);
+            $job->setDescription($description);
+            $job->setActive($active);
+            $job->setPriority($priority);
+            $job->setCreatedAt(new DateTime());
+            $job->setCompany($company);
+
+            $savedJob = $this->jobsRepository->save($job);
+
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'added' => $savedJob,
+                        'error' => false,
+                    ]
+                ]
+            );
+        } catch (Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * getJobById
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function getJobById(int $id): Response
+    {
+        try {
+            $job = $this->jobsRepository->find($id);
+
```

Comment:
what will be the result of trying to get a job that does not exist in the database?
![image](https://user-images.githubusercontent.com/19302124/207305586-1b70c2e7-73c6-47c9-b54a-ca42d7b8d481.png)

$this->jobsRepository->find($id); can return JobEntity if job found in database and NULL if job not found in the database. If not found we need to stop executing the rest of the code. If not we get the error above because job is NULL and we try   'id' => $job->getId() which is null->getId()
Add this   
`if (is_null($job)) {
                throw new \InvalidArgumentException('Could not find job with id: ' . $id);
            }   `

![image](https://user-images.githubusercontent.com/19302124/207306608-b200aa3d-64e1-4479-ad8d-944150216ab4.png)